### PR TITLE
[ci]: add pidetop (fix #7336)

### LINKED ISSUE #7336: add pidetop to CI
that should also help #6859

### PR DESCRIPTION
**Pull request by gares**
We test pidetop compiles, installs and can be run by coq

### CODE REVIEW

File: dev/ci/ci-pidetop.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+pidetop_CI_DIR="${CI_BUILD_DIR}/pidetop"
+
+git_checkout "${pidetop_CI_BRANCH}" "${pidetop_CI_GITURL}" "${pidetop_CI_DIR}"
+
+( cd "${pidetop_CI_DIR}" && coq_makefile -f Make -o Makefile.top && make -f Makefile.top "-j${NJOBS}" && make install-toploop -f Makefile.top )
+
+# Test the toploop can be loaded by Coq (grep init detects if it was loaded,
+# the || true is there because 'xxx' is not a valid message and we run with -o
+# pipefail)
+echo xxx | (coqtop -main-channel stdfds -toploop pidetop || true) | grep init
```
Comment (ejgallego) [inline dev/ci/ci-pidetop.sh]: I am not sure if I like this test, WDYT @SkySkimmer ?

Comment (gares) [reply dev/ci/ci-pidetop.sh]: It may help to know what is wrong with it, in your opinion.

Comment (SkySkimmer) [reply dev/ci/ci-pidetop.sh]: Is there some reason we shouldn't use a valid message?

**Review by SkySkimmer — APPROVED** [review]
I would prefer to use a valid message for the test if possible.

File: dev/ci/ci-pidetop.sh (hunk 1 above)
Comment (gares) [reply dev/ci/ci-pidetop.sh]: I can try to see if there is an equivalent of `Quit.` (now that I think at it, there should be a `Terminate` one).

**Comment by ejgallego** [comment]
So what @SkySkimmer said.

**Comment by gares** [comment]
The original protocol has no "exit", exiting is signaled by closing the channel (and not exit code is pecified AFAICT).
I've added a message "exit" that calls `exit 0`, so that we can tell if coq exited abruptly or because out message got the expected handler being called.

**Review by ejgallego — APPROVED** [review]
Looks good, will merge as soon as CI is ready.

**Comment by ejgallego** [comment]
Note the backporting does need to update the branch.

**Comment by Zimmi48** [comment]
There is no `v8.8` branch and this wouldn't work with the new `exit` message anyways, so I suppose we'll just test this in 8.9 and forwards.

**Comment by ejgallego** [comment]
:+1:

**Pull request merged**